src/LanguageManager/LanguageUsageList.php

```diff
@@ -0,0 +1,84 @@
+<?php # -*- coding: utf-8 -*-
+
+declare( strict_types = 1 );
+
+namespace Inpsyde\MultilingualPress\LanguageManager;
+
+use Inpsyde\MultilingualPress\API\Languages;
+use Inpsyde\MultilingualPress\Common\Type\Language;
+use function Inpsyde\MultilingualPress\get_available_languages;
+
+class LanguageUsageList {
+
+	const ACTIVE   = 1;
+	const INACTIVE = 2;
+
+	/**
+	 * @var \Inpsyde\MultilingualPress\API\Languages
+	 */
+	private $languages;
+
+	/**
+	 * Languages separated by their usage status.
+	 *
+	 * @var array
+	 */
+	private $separated = [];
+
+	/**
+	 * LanguageUsageList constructor.
+	 *
+	 * @param \Inpsyde\MultilingualPress\API\Languages $languages
+	 */
+	public function __construct( Languages $languages )
+	{
+		$this->languages = $languages;
+	}
+
+	/**
+	 * Get either active or inactive languages.
+	 *
+	 * @param int $type
+	 *
+	 * @return array
+	 */
+	public function get_by( int $type ) : array
+	{
+		if ( empty ( $this->separated ) ) {
+			$this->separate();
+		}
+
+		return $this->separated[ $type ] ?? [];
+	}
+
+	/**
+	 * Separates languages by their usage status
+	 *
+	 * @return void
+	 */
+	private function separate()
+	{
+		// Prepare array. Once this has been done, $this->separated cannot be
+		// empty anymore, so get_by() won't run this twice, even if there are
+		// no languages available at all.
+		$this->separated = [ self::ACTIVE => [], self::INACTIVE => [] ];
```

Comment:
Why?

Reply:
Not if there are no languages.

Reply:
Sure, it'd be an [empty array](https://3v4l.org/OcdRj) then, which is no problem because the getter is using null calescence.

Reply:
hmm, the point is that I have this check:

```
		if ( empty ( $this->separated ) ) {
			$this->separate();
		}
```
That shouldn't run twice in case of missing languages. But I can also store an `already_checked` value in the class. Your choice? :)

Reply:
Well, if you remove the initial value, as suggested 😉, then you can just use `if ( ! isset( ) )`. After the first (and only) run of the method, it will be an array, with or without contents.